contracts/strategy/BnbxYieldConverterStrategy.sol

```diff
@@ -0,0 +1,245 @@
+    function _deposit(uint256 amount)
+        internal
+        whenDepositNotPaused
+        returns (uint256)
+    {
+        require(canDeposit(amount), "invalid amount");
+
+        _bnbDepositBalance += amount;
+        _bnbxHoldingBalance += _stakeManager.convertBnbToBnbX(amount);
+        _stakeManager.deposit{value: amount}();
+        return amount;
+    }
+
+    /// @dev withdraws the given amount of BNB from Stader's stakeManager
+    /// @param amount amount of BNB to withdraw
+    function withdraw(address recipient, uint256 amount)
+        external
+        onlyVault
+        returns (uint256)
+    {
+        return _withdraw(recipient, amount);
+    }
+
+    /// @dev withdraws everything from Stader's stakeManager
+    function panic() external onlyStrategist returns (uint256) {
+        (, , uint256 debt) = _vault.strategyParams(address(this));
+        return _withdraw(address(_vault), debt);
+    }
+
+    /// @dev internal function to withdraw the given amount of BNB from Stader's stakeManager
+    /// @param amount amount of BNB
+    /// @return value - returns the amount of BNB withdrawn from stader (currently 0 as no amount is withdrawn in this step)
+    function _withdraw(address recipient, uint256 amount)
+        internal
+        returns (uint256 value)
+    {
+        require(amount > 0, "invalid amount");
+
+        uint256 bnbxAmount = _stakeManager.convertBnbToBnbX(amount);
+        _bnbDepositBalance -= amount;
+        _bnbxHoldingBalance -= bnbxAmount;
+
+        _bnbxToUnstake += bnbxAmount;
+        _withdrawRequests[_nextWithdrawIdx++] = UserWithdrawRequest({
+            recipient: recipient,
+            amount: amount,
+            bnbxAmount: bnbxAmount,
+            triggerTime: block.timestamp
+        });
+
+        return 0;
+    }
+
+    // actual withdraw request to stader
+    function batchWithdraw() external {
+        require(_bnbxToUnstake > 0, "No BNBx to unstake");
+
+        uint256 bnbxToUnstake = _bnbxToUnstake; // To prevent reentrancy
+        _bnbxToUnstake = 0;
+        _stakeManager.requestWithdraw(bnbxToUnstake);
+    }
+
+    /// @dev claims the next available withdraw batch from stader
+    /// @dev transfer funds(BNB) from stakeManager to strategy
+    /// @return foundClaimableReq : true if claimed any batch, false if no batch is available to claim
+    function claimNextBatch() external returns (bool foundClaimableReq) {
+        uint256 claimableIdx;
+        bool foundClaimableReq = false;
+
+        IStakeManager.WithdrawalRequest[] memory requests = _stakeManager
+            .getUserWithdrawalRequests(address(this));
+
+        for (uint256 idx = 0; idx < requests.length; idx++) {
+            (bool isClaimable, uint256 amount) = _stakeManager
+                .getUserRequestStatus(address(this), idx);
+            if (isClaimable) {
+                foundClaimableReq = true;
+                claimableIdx = idx;
+                _bnbToDistribute += amount;
```

Comment:
Why not move stakeManager.claimWithdraw(idx) here? 

Return true instead of break.

Return false at the end of fn.

Reply:
You mean, 
after funds are sent back to this strategy contract, we will transfer it to stader treasury?

It would be better to handle in our contract (`stakeManager`)?

Reply:
Nevermind this.

Reply:
sure